SearchBar/src/Components/SearchBar/VisiblePosts.jsx

```diff
@@ -0,0 +1,36 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+export default function VisiblePosts({ allPosts, query }) {
+  const filterPosts = (posts, searchQuery) => {

```

Comment:
смотри, здесь отличное место, чтобы использовать useMemo хук. Он возвращает "мемоизированное" значение. нужно в него передать функцию и массив зависимостей и useMemo будет пересчитывать значение только тогда, когда одна из зависимостей изменилась. 
Эта хорошая оптимизация, чтобы избежать доп вычислений при каждом рендеринге.

Выглядеть у тебя будет вот так
```suggestion
	const posts = useMemo(() => {
		let filteredPosts = allPosts;

		if (query) {
			filteredPosts = allPosts.filter(post => {
				const postName = post.name.toLowerCase();

				return postName.includes(query);
			});
		}

		return filteredPosts.map(post => (
			<li key={post.id}>{post.name}</li>
		));
	}, [allPosts, query]);

```

и тогда в рендеринге можно заменить 
```
{filteredPosts.map(post => (
      <li key={post.id}>{post.name}</li>
 ))}
```

на `{posts}`